apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -32,7 +32,7 @@ defmodule Aecore.Structures.SignedTx do
   @spec validate(SignedTx.t()) :: boolean()
   def validate(tx) do
     case tx.data do
-      %SpendTx{} -> SpendTx.validate(tx)
+      %SpendTx{} -> SpendTx.validate(tx.data) && Keys.verify_tx(tx)
```

Comment:
good for now, with more transactions we should move it to after the `case ... end`